Tests/Marketplace/marketplace_services.py

```diff
@@ -1512,27 +1533,60 @@ def format_metadata(self, user_metadata, pack_content_items, integration_images,
         finally:
             return task_status
 
-    def _get_pack_creation_date(self, index_folder_path):
-        """ Gets the pack created date.
+    def _get_changelog(self, index_folder_path):
+        """ Gets the pack changelog data.
         Args:
             index_folder_path (str): downloaded index folder directory path.
         Returns:
-            datetime: Pack created date.
-
+            dict: Get the changelog from downloaded index
         """
-
-        # load changelog from downloaded index
         changelog_index_path = os.path.join(index_folder_path, self._pack_name, Pack.CHANGELOG_JSON)
         changelog = {}
         if os.path.exists(changelog_index_path):
             with open(changelog_index_path, "r") as changelog_file:
                 changelog = json.load(changelog_file)
+        return changelog
+
+    def _get_pack_creation_date(self, index_folder_path):
+        """ Gets the pack created date.
+        Args:
+            index_folder_path (str): downloaded index folder directory path.
+        Returns:
+            datetime: Pack created date.
+        """
 
+        changelog = self._get_changelog(index_folder_path)
         initial_changelog_version = changelog.get(Pack.PACK_INITIAL_VERSION, {})
+
+        if not initial_changelog_version:
+            # if the changelog exists but the first version isn't PACK_INITIAL_VERSION
+            sorted_changelog = collections.OrderedDict(sorted(changelog.items()))
+            for version, _ in sorted_changelog.items():
+                initial_changelog_version = changelog.get(version, {})
+                break
+
         init_changelog_released_date = initial_changelog_version.get('released',
                                                                      datetime.utcnow().strftime(Metadata.DATE_FORMAT))
         return init_changelog_released_date
 
+    def _get_pack_update_date(self, index_folder_path, pack_was_modified):
+        """ Gets the pack update date.
+        Args:
+            index_folder_path (str): downloaded index folder directory path.
+            pack_was_modified (bool): whether the pack was modified or not.
+        Returns:
+            datetime: Pack update date.
+        """
+        changelog = self._get_changelog(index_folder_path)
+        latest_changelog_released_date = datetime.utcnow().strftime(Metadata.DATE_FORMAT)
+
+        if changelog.keys() and not pack_was_modified:
```

Comment:
```suggestion
        if changelog and not pack_was_modified:
```

Reply:
In what cases `changelog` can be empty? not sure why what's the right behavior in that case

Reply:
If it's a new pack, then it _could_ be empty.

Reply:
But then `pack_was_modified` will be set to True, no?

Reply:
I don't believe so. `pack_was_modified` should only be used when a pack is being re-uploaded. A new pack is a fresh commit so there wouldn't be an existing version to modify